src/capwap_http_api_v2_handler.erl

```diff
@@ -0,0 +1,103 @@
+%% Copyright (C) 2013-2019, Travelping GmbH <info@travelping.com>
+
+%% This program is free software: you can redistribute it and/or modify
+%% it under the terms of the GNU Affero General Public License as published by
+%% the Free Software Foundation, either version 3 of the License, or
+%% (at your option) any later version.
+
+%% This program is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%% GNU Affero General Public License for more details.
+
+%% You should have received a copy of the GNU Affero General Public License
+%% along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-module(capwap_http_api_v2_handler).
+
+-export([handle/4]).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+handle(<<"GET">>, [<<"station">>], Req, State) ->
+	Stations0 = capwap_station_reg:list_stations(),
+	Stations1 = lists:foldl(fun({AC, MAC}, Acc) ->
+					orddict:update(AC, fun(V) -> [MAC|V] end, [MAC], Acc)
+				end, orddict:new(), Stations0),
+	Stations = lists:map(fun({K, V}) ->
+					[WTP|_] = capwap_wtp_reg:get_commonname(K),
+				{WTP, V}
+			     end, orddict:to_list(Stations1)),
+	Res = lists:foldl(fun format_wtps/2, [], Stations),
+	{jsx:encode(Res), Req, State};
+
+handle(<<"DELETE">>, [<<"wtp">>, _WTP, <<"station">>, Station], Req, State) ->
+	case capwap_tools:mac_to_hex(Station) of
+		MACSta when is_binary(MACSta) ->
+			case capwap:detach_station(MACSta) of
+				ok ->
+					{true, Req, State};
+				_ ->
+					{false, Req, State}
+			end;
+		_ ->
+			{false, Req, State}
+	end;
+
+handle(_Method, _Path, Req, State) ->
+	{jsx:encode([{error, bad_command}]), Req, State}.
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+
+format_wtps({{WTP, _WtpIP}, STAs}, Acc0) ->
+	Radios = lists:foldl(fun format_radios/2, #{}, STAs),
+
+	ResRadios = maps:fold(fun(RadioId, STAs0, Acc) ->
+				[[{'radio-id', RadioId},
+				  {stations, STAs0}] | Acc]
+			      end, [], Radios), [[{'wtp-id', WTP},
```

Comment:
line break after the end of the fold function is missing, makes it very hard to read